dump/json.go

```diff
@@ -35,16 +41,36 @@ func PrintChunkFileInJSON(path string, indented bool, writer io.Writer) error {
 			return fmt.Errorf("corrupted fluent-bit chunk file %s on the %dth record: %w", path, lastI, iterError)
 		}
 		return nil
-	}
-
-	// For Fluentd forward message
-	reader := bytes.NewReader(fileData)
-	var message forwardprotocol.Message
-	if decErr := msgpack.NewDecoder(reader).Decode(&message); decErr != nil {
-		return fmt.Errorf("failed to decode forward message file %s: %w", path, decErr)
-	}
-	if prtError := PrintMessageInJSON(message, indented, writer); prtError != nil {
-		return fmt.Errorf("failed to print %s: %w", path, prtError)
+	case ".ff":
+		var message forwardprotocol.Message
+		if decErr := msgpack.NewDecoder(bytes.NewReader(fileData)).Decode(&message); decErr != nil {
+			return fmt.Errorf("failed to decode forward message file %s: %w", path, decErr)
+		}
+		if prtError := PrintMessageInJSON(message, indented, writer); prtError != nil {
+			return fmt.Errorf("failed to print %s: %w", path, prtError)
+		}
+	default:
+		genericReader := bytes.NewReader(fileData)
+		genericDecoder := msgpack.NewDecoder(genericReader)
+		for {
+			nextItem, nextErr := genericDecoder.DecodeInterface()
+			if nextErr != nil {
+				return fmt.Errorf("failed to decode msgpack at %d: %w", genericReader.Size() - int64(genericReader.Len()), nextErr)
+			}
+			var jbin []byte
+			var jerr error
+			if indented {
+				jbin, jerr = json.MarshalIndent(nextItem, "", "  ")
+			} else {
+				jbin, jerr = json.Marshal(nextItem)
+			}
+			if jerr != nil {
+				return fmt.Errorf("failed to format JSON: %w", jerr)
+			}
+			_, _ = writer.Write(jbin)
+			_, _ = writer.Write([]byte("\n"))
+			logger.Debugf("completed decoding at position %d", genericReader.Size() - int64(genericReader.Len()))
+		}
```

Comment:
So essentially you only have two conditions that break the loop and both return an error, so this function will never exit normally I guess? Is it fine?